src/gui/qgsbrowserdockwidget.h

```diff
@@ -102,6 +102,8 @@ class GUI_EXPORT QgsBrowserDockWidget : public QgsDockWidget, private Ui::QgsBro
     void openFile( const QString & );
     //! Emitted when drop uri list needs to be handled
     void handleDropUriList( const QgsMimeDataUtils::UriList & );
+    //! Connections changed in the browser
+    void connectionsChanged( );
```

Comment:
@wonder-sk thanks for reviewing, I thinks this is a leftover from when I tried to make this work the other way round: refresh the dialogs when the browser panel changes (and hence it can be removed). 

As you noticed (and I mentioned in the PR introduction) this implementation only works one way: changes in the dialogs are propagated to the browser(s), and it does also work from one browser to another. 

I tried to find a good way to propagate the changes from the browsers to the provider's dialogs without success. 

I think we should merge this PR (after the changes you requested) and target the synchronization of the dialogs from the browser in a future work.